src/main/java/micdoodle8/mods/galacticraft/core/mixinplugin/MixinPlugin.java

```diff
@@ -0,0 +1,110 @@
+package micdoodle8.mods.galacticraft.core.mixinplugin;
+
+import net.minecraft.launchwrapper.Launch;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.lib.tree.ClassNode;
+import org.spongepowered.asm.mixin.extensibility.IMixinConfigPlugin;
+import org.spongepowered.asm.mixin.extensibility.IMixinInfo;
+import ru.timeconqueror.spongemixins.MinecraftURLClassPath;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static micdoodle8.mods.galacticraft.core.mixinplugin.TargetedMod.*;
+
+public class MixinPlugin implements IMixinConfigPlugin {
+
+    private static final Logger LOG = LogManager.getLogger("MicdoodleCore mixins");
+
+    @Override
+    public void onLoad(String mixinPackage) {
+
+    }
+
+    @Override
+    public String getRefMapperConfig() {
+        return null;
+    }
+
+    @Override
+    public boolean shouldApplyMixin(String targetClassName, String mixinClassName) {
+        return true;
+    }
+
+    @Override
+    public void acceptTargets(Set<String> myTargets, Set<String> otherTargets) {
+
+    }
+
+    // This method return a List<String> of mixins. Every mixins in this list will be loaded.
+    @Override
+    public List<String> getMixins() {
+        final boolean isDevelopmentEnvironment = (boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment");
+
+        List<TargetedMod> loadedMods = Arrays.stream(TargetedMod.values())
+                .filter(mod -> mod == VANILLA || mod == FORGE
+                        || (mod.loadInDevelopment && isDevelopmentEnvironment)
+                        || loadJar(mod.jarNameBeginsWith))
+                .collect(Collectors.toList());
+
+        if(loadedMods.contains(SYNCMOD)) {
+            throw new IllegalStateException("Sync Mod is not supported in this custom version of Galacticraft!");
+        }
+        LOG.info("Hint: Bukkit is not supported in this custom version of Galacticraft!");
+
+        for (TargetedMod mod : TargetedMod.values()) {
+            if(loadedMods.contains(mod)) {
+                LOG.info("Found " + mod.modName + "! Integrating now...");
+            }
+            else {
+                LOG.info("Could not find " + mod.modName + "! Skipping integration....");
+            }
+        }
+
+        List<String> mixins = new ArrayList<>();
+        for (Mixin mixin : Mixin.values()) {
+            if (mixin.shouldLoad(loadedMods)) {
+                mixins.add(mixin.mixinClass);
+                LOG.debug("Loading mixin: " + mixin.mixinClass);
+            }
+        }
+        return mixins;
+    }
+
+    private boolean loadJar(final String jarNameBeginsWith) {
+        try {
+            File jar = MinecraftURLClassPath.getJarInModPath(jarNameBeginsWith);
```

Comment:
There should be improved version of this (if I didn't forget to "merge" it from visual prospecting), which properly matches prefixes and is case insensetive

Reply:
you will need a new version of `TargetedMod` too